src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -158,7 +170,12 @@ public static Uri GetUriFromFilePath(string filePath)
             if (filePath is null)
                 throw new ArgumentNullException(nameof(filePath));
 
-            return new Uri(filePath, UriKind.Absolute);
+            if (Uri.TryCreate(filePath, UriKind.Absolute, out var uri))
+            {
+                return uri;
+            }
+
+            return new Uri(filePath, UriKind.Relative);
```

Comment:
So why are we getting relative URIs here? Is this just the source generated document case or do we have something else going on? I kinda wish there was a comment either way.